app/services/katello/pulp3/smart_proxy_mirror_repository.rb

```diff
@@ -73,6 +73,27 @@ def self.orphan_distribution?(distribution)
             distribution.try(:repository_version).nil?
       end
 
+      def delete_orphan_alternate_content_sources
+        tasks = []
+        known_acs_hrefs = []
+        known_acss = smart_proxy.smart_proxy_alternate_content_sources
+        known_acs_hrefs = known_acss.pluck(:alternate_content_source_href) if known_acss.present?
+
+        file_acs_api = ::Katello::Pulp3::Repository.api(smart_proxy, 'file').alternate_content_source_api
+        yum_acs_api = ::Katello::Pulp3::Repository.api(smart_proxy, 'yum').alternate_content_source_api
+
+        orphan_file_acs_hrefs = file_acs_api.list.results.map(&:pulp_href) - known_acs_hrefs
+        orphan_yum_acs_hrefs = yum_acs_api.list.results.map(&:pulp_href) - known_acs_hrefs
```

Comment:
Could you add a nil check here for known_acs_hrefs.

Reply:
I cover this by setting `known_acs_hrefs` equal to `[]` on line 78.  Then on line 80, I only set `known_acs_hrefs` if there are any known_acss.